docs/DeveloperGuide.adoc

```diff
@@ -298,6 +298,34 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 image::UndoRedoActivityDiagram.png[width="650"]
 
+=== Filter feature
+==== Current Implementation
+
+The filter mechanism is facilitated by `VersionedAddressBook`.
+Given below is an example usage scenario and how the filter mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `VersionedAddressBook`
+will be initialized with the initial address book state.
+
+//image::UndoRedoStartingStateListDiagram.png[width="800"]
```

Comment:
Hey! You can remove line 310 and ..
Otherwise, all good from what can be seen here.